### DISCUSSION TITLE
同一个目标类型，多个转换问题

### DISCUSSION THREAD

**Discussion by curryzxh** [looly/hutool#2896] [General]
ConverterRegistry能否添加一个自定义名称的转换器，有时候会存在同一个目标类型，多个转换器的问题，如下

ConverterRegistry putCustom(Type type,String name,Class<? extends Converter<?>> converterClass)
或者
ConverterRegistry putCustom(String name,Class<? extends Converter<?>> converterClass)

**Comment by looly** [comment]
嗯，确实有这个问题。

不过自定义转换的话，可能你要重写对应的XXXXConverter了。

因为正式使用时，都是自动转换，没法指定名称，既然指定了名称，为何不自己转？

**Reply by curryzxh** [reply]
为了统一转换框架使用形式
转换框架大同小异，因为这些小的差异，又换自己的框架，舍弃现有框架，有点重复造轮子。
所以希望能够有一定灵活度是最好的。
